### PR TITLE
kibana: add CreateFleetServerHosts and CreateFleetProxy

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

Adds `kibana.CreateFleetProxy` and `kibana.CreateFleetServerHosts`

## Why is it important?

Those kibana APIs are needed for the tests added on https://github.com/elastic/elastic-agent/pull/5889

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works

## Related issues

- Relates https://github.com/elastic/elastic-agent/pull/5889

### CODE REVIEW

File: kibana/fleet.go [hunk 1]
```diff
@@ -474,21 +474,43 @@ func (client *Client) UpgradeAgent(ctx context.Context, request UpgradeAgentRequ
...
-	IsDefault       bool     `json:"is_default"`
 	HostURLs        []string `json:"host_urls"`
+	IsDefault       bool     `json:"is_default"`
+	IsInternal      bool     `json:"is_internal"`
 	IsPreconfigured bool     `json:"is_preconfigured"`
+	ProxyID         string   `json:"proxy_id"`
 }
 
-// ListFleetServerHostsRequest is currently unused
+// ListFleetServerHostsRequest ...
```
Comment (rdner) [inline kibana/fleet.go:484]: I think we need a real comment here now.

File: kibana/fleet.go [hunk 2]
```diff
@@ -502,6 +524,35 @@ func (client *Client) ListFleetServerHosts(ctx context.Context, _ ListFleetServe
 	return r, err
 }
 
+// CreateFleetServerHosts creates a new Fleet Server host
+func (client *Client) CreateFleetServerHosts(ctx context.Context, req ListFleetServerHostsRequest) (FleetServerHostsResponse, error) {
+	bs, err := json.Marshal(req)
+	if err != nil {
+		return FleetServerHostsResponse{}, fmt.Errorf("could not marshal ListFleetServerHostsRequest")
```
Comment (rdner) [inline kibana/fleet.go:531]:
```suggestion
		return FleetServerHostsResponse{}, fmt.Errorf("could not marshal ListFleetServerHostsRequest: %w", err)
```

File: kibana/fleet.go [hunk 3]
```diff
@@ -628,6 +683,59 @@ func (client *Client) DeleteFleetPackage(ctx context.Context, packagePolicyID st
 	return r, err
 }
 
+//
+// Fleet Proxies
+//
+
+type ProxiesRequest struct {
```
Comment (rdner) [inline kibana/fleet.go:690]: Should we add links to the docs like for the rest of the structs?

File: kibana/fleet.go [hunk 4]
```diff
@@ -628,6 +683,59 @@ func (client *Client) DeleteFleetPackage(ctx context.Context, packagePolicyID st
...
+	CertificateAuthorities string            `json:"certificate_authorities"`
+	CertificateKey         string            `json:"certificate_key"`
+	ID                     string            `json:"id"`
+	IsPreconfigured        bool              `json:"is_preconfigured"`
+	Name                   string            `json:"name"`
+	ProxyHeaders           map[string]string `json:"proxy_headers"`
+	URL                    string            `json:"url"`
+}
+
+type ProxiesResponse struct {
```
Comment (rdner) [inline kibana/fleet.go:701]: Link to the docs?

File: kibana/fleet.go [hunk 5]
```diff
@@ -628,6 +683,59 @@ func (client *Client) DeleteFleetPackage(ctx context.Context, packagePolicyID st
...
+		ProxyHeaders           map[string]string `json:"proxy_headers"`
+		URL                    string            `json:"url"`
+	} `json:"item"`
+}
+
+// CreateFleetProxy creates a new proxy
+func (client *Client) CreateFleetProxy(ctx context.Context, req ProxiesRequest) (ProxiesResponse, error) {
+	bs, err := json.Marshal(req)
+	if err != nil {
+		return ProxiesResponse{}, fmt.Errorf("could not marshal ListFleetServerHostsRequest")
```
Comment (rdner) [inline kibana/fleet.go:718]:
```suggestion
		return ProxiesResponse{}, fmt.Errorf("could not marshal ListFleetServerHostsRequest: %w", err)
```

File: kibana/fleet.go [hunk 6]
```diff
@@ -628,6 +683,59 @@ func (client *Client) DeleteFleetPackage(ctx context.Context, packagePolicyID st
...
+	)
+	if err != nil {
+		return ProxiesResponse{}, err
+	}
+	defer r.Body.Close()
+
+	resp := ProxiesResponse{}
+	err = readJSONResponse(r, &resp)
+	if err != nil {
+		return ProxiesResponse{}, fmt.Errorf("failes parsing response")
```
Comment (rdner) [inline kibana/fleet.go:733]:
```suggestion
		return ProxiesResponse{}, fmt.Errorf("failes parsing response: %w", err)
```

**Review by rdner — COMMENTED** [review]
It looks like we never check HTTP status codes, is there any reason not to do that?

File: kibana/fleet.go (hunk 3 above)
Comment (AndersonQ) [reply kibana/fleet.go:659]: Kinda, there isn't a standard in the file. So I created one, removing those redundant comments and linking to the API doc on the comment marking the methods about the API. Like [this](https://github.com/elastic/elastic-agent-libs/pull/246/files#diff-7e6d55d12ca5e0be223823cc8a7d5c460b540f9cafe565d1052b0a92e3fb39d8R654-R657).

File: kibana/fleet.go (hunk 4 above)
Comment (AndersonQ) [reply kibana/fleet.go:670]: see above

**Review by rdner — APPROVED** [review]
LGTM but we should decide on HTTP status code checks. I'm fine with a follow up PR for this too.

**Pull request merged**